### ISSUE TITLE
Add the domain parameter to config

### ISSUE THREAD

**Issue by nicbarker** [koajs/session#140]
Hi,
It would be very convenient for me if koa-session supported the `domain` directive in it's config.
I'm happy to implement and PR if neccessary.

**Comment by traubisoda** [comment]
Hi,

I had to use the domain directive too, so I dug up the code, and since all config params get passed down to "lower layers", I tried to add a domain key to the config object, and it worked! :)

**Comment by traubisoda** [comment]
> Hi,
> It would be very convenient for me if koa-session supported the `domain` directive in it's config.
> I'm happy to implement and PR if neccessary.

Hi,

I had to use the domain directive too, so I dug up the code, and since all config params get passed down to "lower layers", I tried to add a domain key to the config object, and it worked! :)

**Comment by nicbarker** [comment]
> Hi,
> 
> I had to use the domain directive too, so I dug up the code, and since all config params get passed down to "lower layers", I tried to add a domain key to the config object, and it worked! :)

Huh, that's interesting. I couldn't get it to work for me, although other parameters that are listed as working seemed to work (`signed`, `httpOnly` etc). Maybe I had something else wrong in my configuration.

**Comment by geritol** [comment]
see https://github.com/koajs/session#options

> All other options are passed to ctx.cookies.get() and ctx.cookies.set() allowing you to control security, domain, path, and signing among other settings.

**Comment by geritol** [comment]
the problem is, for typescript users, that `interface opts` does not allow to add other options.

**Comment by dead-horse** [comment]
koa-session didn't provide typescript definition, maybe you can create a pull request to the tsd which you are using?

**Comment by geritol** [comment]
Yes, forgot to mention that this is an issue with `@types/koa-session` and not with this library.

**Comment by rahulgi** [comment]
This seems to be working fine now and can probably be closed.
